File: todolist/src/types/emotion.ts
export interface EmotionEntry {
  id: string;
  date: string;
  emoji_1: string;
  emoji_2: string;
  emoji_3: string;
  emoji_4: string;
  emoji_5: string;
  control_score: number;
  notes: string;
  weather: string;
  task_completed_count: number;
  created_at: string;
  updated_at: string;
}

export interface SaveEmotionParams {
  date: string;
  emoji_1: string;
  emoji_2: string;
  emoji_3: string;
  emoji_4: string;
  emoji_5: string;
  control_score: number;
  notes: string;
  weather: string;
  task_completed_count: number;
}

export interface EmotionHeatmapCell {
  date: string;
  control_score: number;
  emoji_1: string;
}

File: todolist/src-tauri/src/models/emotion.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EmotionEntry {
    pub id: String,
    pub date: String,
    pub emoji_1: String,
    pub emoji_2: String,
    pub emoji_3: String,
    pub emoji_4: String,
    pub emoji_5: String,
    pub control_score: i32,
    pub notes: String,
    pub weather: String,
    pub task_completed_count: i32,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct SaveEmotionParams {
    pub date: String,
    pub emoji_1: String,
    pub emoji_2: String,
    pub emoji_3: String,
    pub emoji_4: String,
    pub emoji_5: String,
    pub control_score: i32,
    pub notes: String,
    pub weather: String,
    pub task_completed_count: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct EmotionHeatmapCell {
    pub date: String,
    pub control_score: i32,
    pub emoji_1: String,
}
